ctr-nonce-reuse/src/ctr_game.rs

```diff
@@ -0,0 +1,196 @@
+    key: [u8; BLOCK_SIZE],
+    nonce: [u8; BLOCK_SIZE],
+
+    known_plaintext: Vec<u8>,
+    flag_plaintext: Vec<u8>,
+
+    plaintexts_vec: Vec<Vec<u8>>,
+    ciphertexts_vec: Vec<Vec<u8>>,
+}
+
+impl Default for CtrGame {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CtrGame {
+    pub fn new() -> Self {
+        let rng = rand::thread_rng();
+        let mut init_rng = rng.clone();
+        //let key:[u8; BLOCK_SIZE] = rng.gen();
+        //let nonce:[u8; BLOCK_SIZE] = rng.gen();
+
+        Self {
+            rng,
+
+            //aes_ctr: Aes128Ctr64LE::new(&key.into(), &nonce.into()),
+            key: init_rng.gen(),
+            nonce: init_rng.gen(),
+
+            known_plaintext: vec![],
+            flag_plaintext: vec![],
+
+            plaintexts_vec: vec![
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "M"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Tom",  "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "Administrator", "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                //br#"{ "Heartbeat": "Alive",         "Since": "1653924358" }"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Alex", "Surname": "M"}"#.to_vec(),
+                br#"{  "UserType": "UsualUser",     "Name": "Tom",  "Surname": "S"}"#.to_vec(),
+                br#"{  "UserType": "Administrator", "Name": "Alex", "Surname": "S"}"#.to_vec(),
+            ],
+            ciphertexts_vec: vec![],
+        }
+    }
+
+    pub fn restart(&mut self) {
+        self.rng = rand::thread_rng();
+
+        self.key = self.rng.gen();
+        self.nonce = self.rng.gen();
+        //self.aes_ctr = Aes128Ctr64LE::new(&key.into(), &nonce.into());
+
+        let flag_str = self.create_random_flag();
+        self.flag_plaintext = flag_str.into();
+
+        self.create_ciphertexts(EAVESDROPPED_MSG_COUNT);
+    }
+
+    fn create_random_flag(&mut self) -> String {
+        let left_random_bytes: [u8; 4] = self.rng.gen();
+        let right_random_bytes: [u8; 4] = self.rng.gen();
+        let mut flag_str = String::new();
+
+        write!(
+            flag_str,
+            "FLAG{{{}_CtR_1sNT_SeCC_{}}}",
+            Self::byte_to_hex(&left_random_bytes),
+            Self::byte_to_hex(&right_random_bytes)
+        )
+        .expect("String formatting can not fail");
+
+        flag_str
+    }
+
+    fn create_ciphertexts(&mut self, count: usize) {
+        self.ciphertexts_vec.clear();
+        let mut chosen_plaintext: Vec<_> = self
+            .plaintexts_vec
+            .choose_multiple(&mut self.rng, count)
+            .collect();
+        self.known_plaintext = (*chosen_plaintext.choose(&mut self.rng).expect("a")).clone();
```

Comment:
This star expression with the `clone()` can be replaced by `to_vec()`:
```suggestion
        self.known_plaintext = chosen_plaintext.choose(&mut self.rng).expect("non-empty").to_vec();
```